src/winforms/toga_winforms/widgets/slider.py

```diff
@@ -0,0 +1,39 @@
+from travertino.size import at_least
+from toga_winforms.libs import WinForms
+from toga_winforms import color
+from .base import Widget
+
+
+class TogaSlider(WinForms.TrackBar):
+    def __init__(self, interface):
+        super().__init__()
+        self.interface = interface
+        self.Scroll += self.on_slide
+
+    def on_slide(self, sender, event):
+        if self.interface.on_slide:
+            self.interface.on_slide(self.interface)
+
+
+class Slider(Widget):
+    def create(self):
+        self.native = TogaSlider(self.interface)
+        self.set_enabled(self.interface._enabled)
+
+    def get_value(self):
+        return self.native.Value
+
+    def set_value(self, value):
+        self.native.Value = value
+
+    def set_range(self, range):
+        # TODO: Should I use self.interface.range etc for the values?
```

Comment:
Yes, it should be self.interface.range. The argument passed to `set_range` is the "raw" value, before any error handling or processing. There generally won't be any need to use that value.

Reply:
Edited in winforms, cocoa and gtk.